### ISSUE TITLE
[bug] False positive in @eslint-react/prefer-read-only-props

### ISSUE THREAD

**Issue by boiseiben** [Rel1cx/eslint-react#1122]
### Describe the bug

When passing a state-setting function to a React state setter (so as to make the state update as a function of the current state), under some conditions the parameter to that function is flagged as having an @eslint-react/prefer-read-only-props error, which is very confusing.

### Reproduction

```tsx
import { useState } from 'react';

type Book = {
	title: string,
	author: string,
}
export default function MyComponent() {
	const [books, setBooks] = useState<Book[]>([]);
	const dropFirstBook = () => {
		setBooks(
			// Error in next line: A function component's props should be read-only.
			// Source: @eslint-react/prefer-read-only-props
			(currentBooks: Book[]) => {
				const newBooks: Book[] = [];
				// Considerable additional logic is found here in a non-toy example;
				// suggestions to just rewrite this as e.g.,
				// return [...currentBooks.slice(1)];
				// which gets the above line reported as compliant with the rule
				// are not nearly as helpful as they might seem,
				// though knowing about them might help find the root cause of the bug.
				newBooks.push(...currentBooks.slice(1));
				return newBooks;
			}
		);
	};
	//Please assume there's more that was simplified away for this bug demonstration.
	if(books.length > 0) {
		return(<>
			The first book is {books[0].title}.
			<button
				type='button'
				onClick={dropFirstBook}
			> Remove it. </button>
		</>);
	} else {
		return null;
	}
}
````

### Expected behavior

No error/warning is reported; this rule doesn't flag anything. 

### Platform and versions

```Text
Node v18.19.1

I reinstalled the packages today, so it should be the latest: 

      '@eslint-react/eslint-plugin':
        version: 1.51.0(eslint@9.24.0(jiti@2.4.2))(ts-api-utils@2.1.0(typescript@5.8.3))(typescript@5.8.3)
      '@eslint/eslintrc':
        version: 3.3.1
      '@eslint/js':
        version: 9.28.0
```

### Stack trace

```Text

```

**Comment by Rel1cx** [comment]
Hi @boiseiben! Thanks for the reproduction code and the comments u provided inside, it did help a lot for finding the root cause.

**Comment by Rel1cx** [comment]
The root cause is that our implementation of the heuristic react component detection (distinguishing which should be considered as react components from various forms of JS functions) had a bug caused the `(currentBooks: Book[]) => { ... }` function was incorrectly considered an anonymous React function component. This causes its props to be accidentally checked by this rule.

**Comment by boiseiben** [comment]
Thanks so much for the quick response and fix @Rel1cx!
